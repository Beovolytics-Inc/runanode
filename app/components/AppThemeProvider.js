import React, { Component } from 'react';
import { ThemeProvider } from 'styled-components';

export default class TgeAppThemeProvider extends Component {
  componentWillMount() {
    const { globalStyles } = this.props;
    this.stylesheet = document.createElement('style');
    this.stylesheet.type = 'text/css';
    this.stylesheet.innerHTML = globalStyles;

    if (document && document.head) {
      document.head.appendChild(this.stylesheet);
    }
  }

  componentWillUnmount() {
    if (this.stylesheet && document && document.head) {
      document.head.removeChild(this.stylesheet);
      delete this.stylesheet;
    }
  }

  render() {
    const { theme, children } = this.props;
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
}
