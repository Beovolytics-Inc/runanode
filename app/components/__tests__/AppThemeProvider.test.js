import React from 'react';
import AppThemeProvider from '../AppThemeProvider';

describe('AppThemeProvider', () => {
  describe('render', () => {
    test('should load theme', () => {
      const theme = {};
      const Component = (
        <AppThemeProvider theme={theme}>
          <div>test</div>
        </AppThemeProvider>
      );
      const wrapper = shallow(Component);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
