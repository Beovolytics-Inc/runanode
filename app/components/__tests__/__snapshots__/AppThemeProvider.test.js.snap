// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppThemeProvider render should load theme 1`] = `
<ThemeProvider
  theme={Object {}}
>
  <div>
    test
  </div>
</ThemeProvider>
`;
